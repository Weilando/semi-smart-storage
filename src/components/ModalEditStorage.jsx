import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Modal.css';

function ModalEditStorage(props) {
  ModalEditStorage.propTypes = {
    show: PropTypes.bool.isRequired,
    sName: PropTypes.string.isRequired,
    buffer: PropTypes.string.isRequired,
    changeAction: PropTypes.func.isRequired,
    closeAction: PropTypes.func.isRequired,
    deleteAction: PropTypes.func.isRequired,
    updateAction: PropTypes.func.isRequired,
  }

  return(
    <div className={'modal '.concat(props.show ? 'show' : 'hide')}>
      <div className="modal-main">
        <span className="close" onClick={props.closeAction}>&times;</span>
        <h1>Settings for storage "{props.sName}".</h1>

        <h2>Update name</h2>
        <form onSubmit={props.updateAction} className="Modal">
          <input
            type="text"
            className="Modal"
            value={props.buffer}
            onChange={props.changeAction}
          />
          <br/>
          <input
            type="submit"
            className="Modal"
            value="Update name"
            disabled={props.sName === ''}
          />
        </form>

        <h2>Delete storage</h2>
        <p>To delete the storage and all its content, click the button below.
        Items and units are not deleted.</p>
        <button onClick={props.deleteAction} className="Modal">
          Delete Storage
        </button>
      </div>
    </div>
  );
}

export default ModalEditStorage;
