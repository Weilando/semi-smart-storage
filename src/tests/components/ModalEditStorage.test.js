import React from 'react';
import renderer from 'react-test-renderer';
import ModalEditStorage from '../../components/ModalEditStorage';

describe('snapshot tests for ModalEditStorage component', () => {
  it('should render visible ModalEditStorage component correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const deleteAction = jest.fn();
    const updateAction = jest.fn();


    const tree = renderer.create(
      <ModalEditStorage
        show={true}
        sName="Fridge"
        buffer="Fridgerator"
        closeAction={closeAction}
        changeAction={changeAction}
        deleteAction={updateAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  it('should render hidden ModalEditStorage component correctly', () => {
    const changeAction = jest.fn();
    const closeAction = jest.fn();
    const deleteAction = jest.fn();
    const updateAction = jest.fn();


    const tree = renderer.create(
      <ModalEditStorage
        show={false}
        sName="Fridge"
        buffer="Fridge"
        closeAction={closeAction}
        changeAction={changeAction}
        deleteAction={updateAction}
        updateAction={updateAction}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })
});
