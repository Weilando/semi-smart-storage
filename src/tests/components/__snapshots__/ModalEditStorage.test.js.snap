// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests for ModalEditStorage component should render hidden ModalEditStorage component correctly 1`] = `
<div
  className="modal hide"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Settings for storage "
      Fridge
      ".
    </h1>
    <h2>
      Update name
    </h2>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        onChange={[MockFunction]}
        type="text"
        value="Fridge"
      />
      <br />
      <input
        className="Modal"
        disabled={false}
        type="submit"
        value="Update name"
      />
    </form>
    <h2>
      Delete storage
    </h2>
    <p>
      To delete the storage and all its content, click the button below. Items and units are not deleted.
    </p>
    <button
      className="Modal"
      onClick={[MockFunction]}
    >
      Delete Storage
    </button>
  </div>
</div>
`;

exports[`snapshot tests for ModalEditStorage component should render visible ModalEditStorage component correctly 1`] = `
<div
  className="modal show"
>
  <div
    className="modal-main"
  >
    <span
      className="close"
      onClick={[MockFunction]}
    >
      ×
    </span>
    <h1>
      Settings for storage "
      Fridge
      ".
    </h1>
    <h2>
      Update name
    </h2>
    <form
      className="Modal"
      onSubmit={[MockFunction]}
    >
      <input
        className="Modal"
        onChange={[MockFunction]}
        type="text"
        value="Fridgerator"
      />
      <br />
      <input
        className="Modal"
        disabled={false}
        type="submit"
        value="Update name"
      />
    </form>
    <h2>
      Delete storage
    </h2>
    <p>
      To delete the storage and all its content, click the button below. Items and units are not deleted.
    </p>
    <button
      className="Modal"
      onClick={[MockFunction]}
    >
      Delete Storage
    </button>
  </div>
</div>
`;
